contracts/contracts/lib/LibGateway.sol

```diff
@@ -355,6 +380,19 @@ library LibGateway {
     /// @param arrivingFrom - the immediate subnet from which this message is arriving
     /// @param crossMsg - the cross message to be executed
     function applyMsg(SubnetID memory arrivingFrom, IpcEnvelope memory crossMsg) internal {
+        applyMsg(arrivingFrom, crossMsg, false);
+    }
+    
+    /// @notice executes a cross message if its destination is the current network, otherwise adds it to the postbox to be propagated further
+    /// This function assumes that the relevant funds have been already minted or burnt
+    /// when the top-down or bottom-up messages have been queued for execution.
+    /// This function is not expected to revert. If a controlled failure happens, a new
+    /// cross-message receipt is propagated for execution to inform the sending contract.
+    /// `Call` cross-messages also trigger receipts if they are successful.
+    /// @param arrivingFrom - the immediate subnet from which this message is arriving
+    /// @param crossMsg - the cross message to be executed
+    /// @param expectTopDownOnly - whether the message should be top-down only. Reverts if it is not.
```

Comment:
I think I was misled by the comment here. The idea is `whether the message should be top-down only. Reverts if this flag is true but the crossmsg is actually bottom up`?

Reply:
Reverts if it's both - not only top down :)